libsolidity/formal/SMTChecker.cpp

```diff
@@ -806,12 +824,24 @@ void SMTChecker::arrayIndexAssignment(Assignment const& _assignment)
 	{
 		auto const& varDecl = dynamic_cast<VariableDeclaration const&>(*id->annotation().referencedDeclaration);
 		solAssert(knownVariable(varDecl), "");
+
+		if (varDecl.hasReferenceOrMappingType())
+			resetVariables([&](VariableDeclaration const& _var) {
+				return _var != varDecl && equalNonValueTypes(varDecl, _var);
```

Comment:
Don't you need to reset everything whose type has a suffix that matches the type? I.e. if you have a `uint[][2][] memory x;` and assign to a `uint[][2]`, don't you have to clear `x`?

Reply:
Done